sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/PropagatorFactory.java

```diff
@@ -27,9 +31,41 @@ static PropagatorFactory getInstance() {
   @Override
   public ContextPropagators create(
       PropagatorModel model, SpiHelper spiHelper, List<Closeable> closeables) {
-    List<String> compositeModel = requireNonNull(model.getComposite(), "composite propagator");
-    TextMapPropagator textMapPropagator =
-        TextMapPropagatorFactory.getInstance().create(compositeModel, spiHelper, closeables);
-    return ContextPropagators.create(textMapPropagator);
+    List<TextMapPropagatorModel> textMapPropagatorModels = model.getComposite();
+    Set<String> propagatorNames = new HashSet<>();
+    List<TextMapPropagator> textMapPropagators = new ArrayList<>();
+    if (textMapPropagatorModels != null && !textMapPropagatorModels.isEmpty()) {
+      textMapPropagatorModels.forEach(
+          textMapPropagatorModel -> {
+            TextMapPropagatorAndName propagatorAndName =
+                TextMapPropagatorFactory.getInstance()
+                    .create(textMapPropagatorModel, spiHelper, closeables);
+            textMapPropagators.add(propagatorAndName.getTextMapPropagator());
+            propagatorNames.add(propagatorAndName.getName());
+          });
+    }
+
+    String compositeList = model.getCompositeList();
+    if (compositeList != null) {
+      List<String> propagatorNamesList =
+          // Process string list same as we process OTEL_PROPAGATORS, trimming and filtering empty
+          // and 'none'
+          Arrays.asList(compositeList.split(",")).stream()
+              .map(String::trim)
+              .filter(s -> !s.isEmpty())
+              .filter(s -> !s.equals("none"))
+              .collect(Collectors.toList());
+      for (String propagatorName : propagatorNamesList) {
+        // Only add entries which weren't already previously added
+        if (!propagatorNames.add(propagatorName)) {
+          continue;
+        }
+        textMapPropagators.add(
+            TextMapPropagatorFactory.getPropagator(spiHelper, propagatorName)
+                .getTextMapPropagator());
+      }
```

Comment:
nit: I've seen this in several other places so feel free to ignore this comment, but since the `continue` and the thing below are both one-liners, by inverting the `if` and putting the `add()` into the if, it saves one line and I think is easier to read.

Reply:
Yeah I try generally try to do the style of programming where you check all the conditions and exit / continue, then do the work, with the goal being to to minimize nested code.

But it can go to far, as you point out here.